addons/project/controllers/portal.py

```diff
@@ -542,3 +544,33 @@ def portal_my_task(self, task_id, report_type=None, access_token=None, project_s
             request.session['my_tasks_history'] = task_sudo.ids
         values = self._task_get_page_view_values(task_sudo, access_token, **kw)
         return request.render("project.portal_my_task", values)
+
+    @http.route('/project_sharing/attachment/add_image', type='http', auth='public', methods=['POST'], website=True)
```

Comment:
Since we just want to allow that to authenticate users. 🙂 
```suggestion
    @http.route('/project_sharing/attachment/add_image', type='http', auth='user', methods=['POST'], website=True)
```